Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -0,0 +1,219 @@
+#include "Spire/Ui/OverlayPanel.hpp"
+#include <QApplication>
+#include <QCloseEvent>
+#include <QEvent>
+#include <QGraphicsDropShadowEffect>
+#include <QGuiApplication>
+#include <QHBoxLayout>
+#include <QPropertyAnimation>
+#include <QScreen>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Box.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  const auto DROP_SHADOW_COLOR = QColor(0, 0, 0, 64);
+  const auto DROP_SHADOW_OFFSET = QPoint(0, 3);
+  const auto DROP_SHADOW_RADIUS = 5;
+  const auto DROP_SHADOW_SIZE = 3;
+  const auto FADE_SPEED_MS = 100;
+
+  auto DROP_SHADOW_HEIGHT() {
+    static auto height = scale_height(DROP_SHADOW_SIZE);
+    return height;
+  }
+
+  auto DROP_SHADOW_WIDTH() {
+    static auto width = scale_width(DROP_SHADOW_SIZE);
+    return width;
+  }
+
+  auto DROP_SHADOW_MARGINS() {
+    static auto margins = QMargins(DROP_SHADOW_WIDTH(), DROP_SHADOW_HEIGHT(),
+      DROP_SHADOW_WIDTH(), DROP_SHADOW_HEIGHT());
+    return margins;
+  }
+
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0xFF, 0xFF, 0xFF))).
+      set(border(scale_width(1), QColor::fromRgb(0xA0, 0xA0, 0xA0)));
+    return style;
+  }
+
+  QWidget* find_top_level_window() {
```

Comment:
The function is used to find the top level window. The OverlayPanel moves along with the top level window by tracking the move event. Child widgets don't send the move event, and only top level window can send it, so the OverlayPanel can't receive the move event from its parent but it can get the event from the top level window.

Reply:
Okay, go ahead and remove this function, in general there is no such thing as "the top level window". There are many top level windows and just picking one arbitrarily is incorrect.

Reply:
If the OverlayPanel doesn't get a top-level window, it can't stick on its parent. 

Logically, the top-level window should be the top-level window where the OverlayPanel's parent is located. The top level window can be got by parent->window(). However, if a widget doesn't have a parent, the window() function will return the widget. For example, in the UiViewer, the OverlayPanel's parent is a button, but the button don't specify a parent because the button's parent can't be got from the factory function of the UiProfile. If find_top_level_window is removed, the OverlayPanel can't get move events to move along with its parent.

In general every widget we use should have a parent. If in order to have the "stick" effect and remove this function, we need to add a parent to each widget in the UiViewer.

Reply:
Okay, look at fixing this and then resubmit the PR. Everything else looks good.